plugin/checkout_handlers.go

```diff
@@ -41,6 +47,91 @@ type CheckOutHandler interface {
+type PasswordHandler struct {
+	client secretsClient
+	child  CheckOutHandler
+}
+
+// CheckOut requires no further action from the password handler other than passing along the request.
+func (h *PasswordHandler) CheckOut(ctx context.Context, storage logical.Storage, serviceAccountName string, checkOut *CheckOut) error {
+	return h.child.CheckOut(ctx, storage, serviceAccountName, checkOut)
+}
+
+// CheckIn rotates the service account's password remotely and stores it locally.
+// If this process fails part-way through:
+// 		- An error will be returned.
+//		- The account will remain checked out.
+//		- The client may (or may not) retry the check-in.
+// 		- The overdue watcher will still check it in if its lending period runs out.
+func (h *PasswordHandler) CheckIn(ctx context.Context, storage logical.Storage, serviceAccountName string) error {
+	if err := validateInputs(ctx, storage, serviceAccountName, nil, false); err != nil {
+		return err
+	}
+	// On check-ins, a new AD password is generated, updated in AD, and stored.
+	engineConf, err := readConfig(ctx, storage)
+	if err != nil {
+		return err
+	}
+	if engineConf == nil {
+		return errors.New("the config is currently unset")
+	}
+	newPassword, err := util.GeneratePassword(engineConf.PasswordConf.Formatter, engineConf.PasswordConf.Length)
+	if err != nil {
+		return err
+	}
+	if err := h.client.UpdatePassword(engineConf.ADConf, serviceAccountName, newPassword); err != nil {
+		return err
+	}
+	entry, err := logical.StorageEntryJSON("password/"+serviceAccountName, newPassword)
+	if err != nil {
+		return err
+	}
+	if err := storage.Put(ctx, entry); err != nil {
+		return err
+	}
+	return h.child.CheckIn(ctx, storage, serviceAccountName)
+}
+
+// Delete simply deletes the password from storage so it's not stored unnecessarily.
+func (h *PasswordHandler) Delete(ctx context.Context, storage logical.Storage, serviceAccountName string) error {
+	if err := validateInputs(ctx, storage, serviceAccountName, nil, false); err != nil {
+		return err
+	}
+	if err := storage.Delete(ctx, "password/"+serviceAccountName); err != nil {
+		return err
+	}
+	return h.child.Delete(ctx, storage, serviceAccountName)
+}
+
+// Status doesn't need any password work.
+func (h *PasswordHandler) Status(ctx context.Context, storage logical.Storage, serviceAccountName string) (*CheckOut, error) {
+	return h.child.Status(ctx, storage, serviceAccountName)
+}
+
+// retrievePassword is a utility function for grabbing a service account's password from storage.
+// retrievePassword will return:
+//  - "password", nil if it was successfully able to retrieve the password.
+//  - ErrNotFound if there's no password presently.
+//  - Some other err if it was unable to complete successfully.
+func retrievePassword(ctx context.Context, storage logical.Storage, serviceAccountName string) (string, error) {
+	entry, err := storage.Get(ctx, "password/"+serviceAccountName)
```

Comment:
I realize this is already merged but I'm curious if `"password"` here should be a constant, like `StorageHandler`'s `checkoutStoragePrefix`